pkg/roachprod/install/cockroach.go

```diff
@@ -937,6 +952,62 @@ func (c *SyncedCluster) initializeCluster(
 	return res, err
 }
 
+// createAdminUserForSecureCluster creates a `roach` user with admin
+// privileges. The password used matches the virtual cluster name
+// ('system' for the storage cluster).
+func (c *SyncedCluster) createAdminUserForSecureCluster(
+	ctx context.Context, l *logger.Logger, startOpts StartOpts,
+) (*RunResultDetails, error) {
+	if !c.Secure {
+		return &RunResultDetails{}, nil
+	}
+
+	const username = "roach"
+	// N.B.: we cannot reuse username and password combinations across
+	// the system interface and virtual clusters due to #109691.
+	var password = startOpts.VirtualClusterName
+	if startOpts.VirtualClusterName == "" {
+		password = SystemInterfaceName
+	}
+
+	stmts := strings.Join([]string{
+		fmt.Sprintf("CREATE USER IF NOT EXISTS %s WITH LOGIN PASSWORD '%s'", username, password),
+		fmt.Sprintf("GRANT ADMIN TO %s", username),
+	}, "; ")
+
+	var result *RunResultDetails
+	// We retry a few times here because cockroach process might not be
+	// ready to serve connections at this point. We also can't use
+	// `COCKROACH_CONNECT_TIMEOUT` in this case because that would not
+	// work for shared-process virtual clusters.
+	retryOpts := retry.Options{MaxRetries: 20}
+	if err := retryOpts.Do(ctx, func(ctx context.Context) error {
+		results, err := c.ExecSQL(
+			ctx, l, c.Nodes[:1], startOpts.VirtualClusterName, startOpts.SQLInstance, []string{
+				"-e", stmts,
+			})
+
+		if err != nil || results[0].Err != nil {
+			err := errors.CombineErrors(err, results[0].Err)
+			l.Printf("error (retrying): %v, output: %s", err, results[0].CombinedOut)
+			return err
+		}
+
+		result = results[0]
+		return nil
+	}); err != nil {
+		return &RunResultDetails{}, fmt.Errorf("failed to create default admin user: %w", err)
```

Comment:
given all the weird and unexpected errors encountered while adding the automatic backup schedule to roachprod, I wonder if it's worth merely logging this error.

Reply:
I'll keep as-is until we know it's a problem. Since this is only running for secure clusters, the impact is smaller even if this code were completely broken.